src/app/ngrx/effects/castle.effects.spec.ts

```diff
@@ -199,27 +165,232 @@ describe('CastleEffects', () => {
+        podcastId: episodes[0].podcastId,
+        page: episodes[0].page,
+        guid: episodes[0].guid,
         metricsType: METRICSTYPE_DOWNLOADS,
         interval: INTERVAL_DAILY,
         beginDate: new Date(),
         endDate: new Date()
       }
     };
     const success = new ACTIONS.CastleEpisodeMetricsSuccessAction({
-      seriesId: podcasts[0].seriesId,
-      page: episode.page,
-      id: episode.id,
-      guid: episode.guid,
+      podcastId: episodes[0].podcastId,
+      page: episodes[0].page,
+      guid: episodes[0].guid,
       metricsPropertyName: getMetricsProperty(INTERVAL_DAILY, <MetricsType>METRICSTYPE_DOWNLOADS),
-      metrics: downloads
+      metrics: ep0Downloads
     });
 
     actions$.stream = hot('-a', { a: action });
     const expected = cold('-r', { r: success });
     expect(effects.loadEpisodeMetrics$).toBeObservable(expected);
   });
+
+  it('should startWith loading page 1 of podcasts', () => {
+   const success = new ACTIONS.CastlePodcastPageSuccessAction({
+      podcasts: [podcast],
+      page: 1,
+      total: 1,
+      all: true
+    });
+
+    actions$.stream = hot('-');
+    const expected = cold('r', { r: success });
+    expect(effects.loadPodcastPage$).toBeObservable(expected);
+  });
+
+  it('should load next page of podcasts', () => {
+    const pageOfPodcasts = new Array(PODCAST_PAGE_SIZE).fill(podcast);
+    const firstPageAction = new ACTIONS.CastlePodcastPageSuccessAction({
+      podcasts: pageOfPodcasts,
+      page: 1,
+      total: pageOfPodcasts.length * 2,
+      all: true
+    });
+    const nextPageLoadAction = new ACTIONS.CastlePodcastPageLoadAction({
+      page: 2,
+      all: true
+    });
+    actions$.stream = hot('-a', { a: firstPageAction });
+    const expected = cold('-r', { r: nextPageLoadAction });
+    expect(effects.loadNextPodcastPage$).toBeObservable(expected);
+  });
+
+  it('should handle lack of podcasts', () => {
+    castle.root.mockItems('prx:podcasts', []);
+    const action = new ACTIONS.CastlePodcastPageLoadAction({page: 1});
+    const completion = new ACTIONS.CastlePodcastPageFailureAction({error: `Looks like you don't have any podcasts.`});
+    actions$.stream = hot('-', { a: action });
+    const expected = cold('r', { r: completion });
+    expect(effects.loadPodcastPage$).toBeObservable(expected);
+  });
+
+  it('should produce failure action if fails to load podcasts', () => {
+    const error = new Error('Whaaaa?');
+    castle.root.mockError('prx:podcasts', error);
+    const action = new ACTIONS.CastlePodcastPageLoadAction({page: 1});
+    const completion = new ACTIONS.CastlePodcastPageFailureAction({error});
+    actions$.stream = hot('-', { a: action });
+    const expected = cold('r', { r: completion });
+    expect(effects.loadPodcastPage$).toBeObservable(expected);
+  });
+
+  describe('loadPodcastsSuccess$ on podcast load navigation', () => {
+    beforeEach(() => {
+      effects.routerParams.podcastId = undefined;
```

Comment:
tests for navigating to a podcast on app load, we don't have anything on the `/` index route, so the app either navigates to the podcast id in local storage or the first podcast returned from page 1 of the loaded podcasts.